### PR TITLE
Fixing dotnet restore error issues

### PR DESCRIPTION
**Pull request by rajkumar42**
printing dotnet_restore.log to console
catching exception when the command times out

### CODE REVIEW

**Comment by rajkumar42** [comment]
@gregg-miskelly @caslan @wesrupert @chuckries

File: src/MICore/MICoreResources.resx [hunk 1]
```diff
@@ -131,7 +131,9 @@
   <data name="Error_DebuggerInitializeFailed_StdErr" xml:space="preserve">
     <value>Unable to establish a connection to {0}. The following message was written to stderr:
 
-{1}</value>
+{1}
+
+See output window for details.</value>
```
Comment (caslan) [inline src/MICore/MICoreResources.resx]: I think this needs to be "Output Window"

File: src/SSHDebugPS/AD7UnixAsyncCommand.cs [hunk 2]
```diff
@@ -143,9 +143,25 @@ private void Close()
...
 
-            _streamingShell.Close();
+                try
+                {
+                    if (_streamingShell != null && _streamingShell.IsOpen)
+                    {
+                        _streamingShell.Close();
+                    }
+
+                    if (_streamingShell != null)
```
Comment (wesrupert) [inline src/SSHDebugPS/AD7UnixAsyncCommand.cs]:
```csharp
_streamingShell?.Dispose();
_streamingShell = null;
```

File: src/SSHDebugPS/AD7UnixAsyncCommand.cs [hunk 3]
```diff
@@ -143,9 +143,25 @@ private void Close()
                     return;
 
                 _isClosed = true;
-            }
 
-            _streamingShell.Close();
+                try
+                {
+                    if (_streamingShell != null && _streamingShell.IsOpen)
```
Comment (wesrupert) [inline src/SSHDebugPS/AD7UnixAsyncCommand.cs]: I think this line can be simplified to `if (_streamingShell?.IsOpen)`.

File: src/SSHDebugPS/AD7UnixAsyncCommand.cs [hunk 4]
```diff
@@ -143,9 +143,25 @@ private void Close()
...
+
+                    if (_streamingShell != null)
+                    {
+                        _streamingShell.Dispose();
+                        _streamingShell = null;
+                    }
+                }
+                catch (ThreadInterruptedException)
+                {
+                    // This can happen if the command failed on a timeout. Say dotnet restore failed because of insufficient permissions etc.
```
Comment (wesrupert) [inline src/SSHDebugPS/AD7UnixAsyncCommand.cs]: This comment can be improved, it doesn't currently indicate that it's okay that we don't try to dispose again.

**Comment by gregg-miskelly** [comment]
:shipit:

File: src/SSHDebugPS/AD7UnixAsyncCommand.cs (hunk 3 above)
Comment (rajkumar42) [inline src/SSHDebugPS/AD7UnixAsyncCommand.cs]: Primitives return values gets converted into nullable when using ?.  
Keeping it as is.

**Pull request merged**